### PR TITLE
[Merged by Bors] - chore: add `induction_eliminator` to `With{Top,Bot,Zero,One}`

### PR DESCRIPTION
**Pull request by eric-wieser**
This is a smaller version of #12605.

This allows `using` to be dropped from `induction`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/WithOne/Defs.lean [hunk 1]
```diff
@@ -127,7 +127,7 @@ def recOneCoe {C : WithOne α → Sort*} (h₁ : C 1) (h₂ : ∀ a : α, C a) :
 -- would automatically get this; right now in Lean 4...I don't
 -- know if it does or not, and I don't know how to check, so
 -- I'll add it manually just to be sure.
-attribute [elab_as_elim] WithZero.recZeroCoe
+attribute [elab_as_elim, induction_eliminator] WithZero.recZeroCoe
```
Comment (kbuzzard) [inline Mathlib/Algebra/Group/WithOne/Defs.lean:130]: Do you now know for sure that `induction_eliminator` is not being added by `to_additive`? If so then you could add this information to the comment above.

**Review by kbuzzard — APPROVED** [review]
bors d+

**Comment by kbuzzard** [comment]
This looks like a great change to me!

File: Mathlib/Algebra/Group/WithOne/Defs.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Group/WithOne/Defs.lean:130]: No, this whole comment can be removed, I just didn't want to scope creep here

**Comment by urkud** [comment]
Does it work with `cases`?

**Comment by eric-wieser** [comment]
More cleanup is needed to make `cases_eliminator` work, so I left that for a future PR.

bors merge

**Pull request closed without merging**